plutus-core/executables/plc/Main.hs

```diff
@@ -179,9 +186,9 @@ runOptimisations (OptimiseOptions inp ifmt outp ofmt mode) = do
 ---------------- Evaluation ----------------
 
 runEval :: EvalOptions -> IO ()
-runEval (EvalOptions inp ifmt printMode timingMode) = do
+runEval (EvalOptions inp ifmt printMode timingMode semvar) = do
   prog <- readProgram ifmt inp
-  let evaluate = Ck.evaluateCkNoEmit PLC.defaultBuiltinsRuntime
+  let evaluate = Ck.evaluateCkNoEmit (PLC.defaultBuiltinsRuntimeForSemanticsVariant semvar)
```

Comment:
We should perhaps get the typechecker to take account of the variant too, but currently the types are the same for all variants.  Would it even be feasible (or desirable) to have different types for different variants?

Reply:
Yeah, currently we're assuming it just affects the dynamic semantics and not the static semantics... which seems reasonable for now.

Reply:
Yeah, currently we're assuming it just affects the dynamic semantics and not the static semantics... which seems reasonable for now.